scripts/gen_app_partitions.py

```diff
@@ -12,24 +12,24 @@
 # when APP_SHARED_MEM is enabled.
 print_template = """
 		/* Auto generated code do not modify */
+		APP_SHARED_ALIGN;
 		. = ALIGN( 1 << LOG2CEIL(data_smem_{0}b_end - data_smem_{0}));
 		data_smem_{0} = .;
 		KEEP(*(SORT(data_smem_{0}*)))
-		. = ALIGN(_app_data_align);
+		APP_SHARED_ALIGN;
 		. = ALIGN( 1 << LOG2CEIL(data_smem_{0}b_end - data_smem_{0}));
 		data_smem_{0}b_end = .;
 """
 linker_start_seq = """
 	SECTION_PROLOGUE(_APP_SMEM_SECTION_NAME, (OPTIONAL),)
 	{
 		APP_SHARED_ALIGN;
-		_image_ram_start = .;
 		_app_smem_start = .;
 """
 
 linker_end_seq = """
-		_app_smem_end = .;
 		APP_SHARED_ALIGN;
+		_app_smem_end = .;
```

Comment:
this needs to go back to its original place. 
We cant risk adding extra alignment. There is a possibility that this can screw up region size calculation.

Reply:
I am afraid I can't understand why we can screw-up here. There won't be extra alignment.
But I suppose i can revert this - it looks equivalent to me, as long as we do not put any symbols after the end of the last app smem section and before _app_smem_end